src/main/java/staffconnect/model/person/Faculty.java

```diff
@@ -0,0 +1,113 @@
+package staffconnect.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static staffconnect.commons.util.AppUtil.checkArgument;
+
+import java.util.Arrays;
+
+/**
+ * Represents a Person's faculty in the staff book.
+ * Guarantees: immutable; is valid as declared in
+ * {@link #isValidFaculty(String)}
+ */
+public class Faculty {
+    public static final String MESSAGE_CONSTRAINTS =
+            "The content should be a String representing a real faculty of NUS";
+
+    /**
+     * For this version, a valid faculty value should match exactly the full name of the facult,
+     * or the value is invalid
+     */
+    public enum FacultyEnum {
```

Comment:
For conciseness Enums don't contain the term enums in them, rather they are nouns by themselves to represent the collection of type of items they contain. Eg: Faculties